scripts/globals/magic.lua

```diff
@@ -2013,25 +2013,30 @@ xi.magic.calculateBuildDuration = function(target, duration, effect, caster)
     if target:isMob() and target:isNM() then
         local buildRes = xi.magic.getEffectResistance(target, effect, true, caster)
 
-        if target:getMod(buildRes) ~= nil then
-            local builtRes = target:getLocalVar(string.format("[RESBUILD]Base_%s", buildRes))
-            local count = target:getLocalVar(string.format("[RESBUILD]Base_%s_Count", buildRes))
+        local buildResMod = target:getMod(buildRes)
 
-            if builtRes == 0 then
-                duration = duration
-            else
-                duration = utils.clamp(duration - (builtRes + 3), 0, 65535) -- Used to add more fidelity to the build. Adding a mod of 30 will be -3 seconds per cast.
-            end
+        if buildResMod ~= nil then
+            local durationDecrease = target:getLocalVar(string.format("[RESBUILD]Base_%s", buildRes))
+            local spellCount = target:getLocalVar(string.format("[RESBUILD]Base_%s_Count", buildRes))
+
+            duration = utils.clamp(duration - durationDecrease, 0, 65535) -- Used to add more fidelity to the build. Adding a mod of 30 will be -3 seconds per cast.
 
-            target:setLocalVar(string.format("[RESBUILD]Base_%s", buildRes), count * 3)
-            target:setLocalVar(string.format("[RESBUILD]Base_%s_Count", buildRes), count + 1)
+            -- find next duration decrease with min of 1 second (MOD <= 10) and max of 100 seconds (MOD >= 1000)
+            local nextDecrease = utils.clamp(math.floor(buildResMod / 10), 1, 100)
+
+            target:setLocalVar(string.format("[RESBUILD]Base_%s", buildRes), durationDecrease + nextDecrease)
+            target:setLocalVar(string.format("[RESBUILD]Base_%s_Count", buildRes), spellCount + 1)
```

Comment:
I don't think this logic is correct. Isn't this going to exponentially increase in base? I think it should be `spellCount * nextDecrease`

Reply:
> I don't think this logic is correct. Isn't this going to exponentially increase the base? I think it should be `spellCount * nextDecrease`

From my understanding will not be exponential because the nextDecrease is based just on the modifier and not on the previous decrease. This also allows the modifier to be changed dynamically without impacting the previously built up resistance.